src/types/module.rs

```diff
@@ -149,6 +150,90 @@ impl PyModule {
         }
     }
 
+    /// Creates and loads a module named `module_name`,
+    /// containing the Python bytecode passed to `data`
+    /// and pretending to live at `file_name`.
+    ///
+    /// <div class="information">
+    ///     <div class="tooltip compile_fail" style="">&#x26a0; &#xfe0f;</div>
+    /// </div><div class="example-wrap" style="display:inline-block"><pre class="compile_fail" style="white-space:normal;font:inherit;">
+    //
+    ///  <strong>Warning</strong>: This will compile and execute code. <strong>Never</strong> pass untrusted code to this function!
+    ///
+    /// </pre></div>
+    ///
+    /// # Errors
+    ///
+    /// Returns `PyErr` if:
+    /// - `data` is not valid Python bytecode.
+    /// - Any Python exceptions are raised while initializing the module.
+    /// - `file_name` or `module_name` cannot be converted to [`CString`]s.
+    ///
+    /// # Example: bundle in a file at compile time with [`include_bytes!`][std::include_bytes]:
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    ///
+    /// # fn main() -> PyResult<()> {
+    /// // This path is resolved relative to this file.
+    /// let bytes = include_bytes!("../../assets/hello_world.pyc");
+    ///
+    /// Python::with_gil(|py| -> PyResult<()> {
+    ///     PyModule::from_pyc(py, bytes, "example.py", "example")?;
+    ///     Ok(())
+    /// })?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// # Example: Load a file at runtime with [`std::fs::read`].
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    ///
+    /// # fn main() -> PyResult<()> {
+    /// // This path is resolved by however the platform resolves paths,
+    /// // which also makes this less portable. Consider using `include_str`
+    /// // if you just want to bundle a script with your module.
+    /// let bytes = std::fs::read("assets/hello_world.pyc")?;
+    ///
+    /// Python::with_gil(|py| -> PyResult<()> {
+    ///     PyModule::from_pyc(py, &bytes, "example.py", "example")?;
+    ///     Ok(())
+    /// })?;
+    /// Ok(())
+    /// # }
+    /// ```
+    pub fn from_pyc<'p>(
```

Comment:
Should this function be `unsafe`? I think despite it being a security hazard I don't think it falls under Rust's typical definition of `unsafe`, so my argument is that it does not need to me.

Reply:
> I don't think it falls under Rust's typical definition of unsafe

Are you sure it cannot trigger undefined behaviour by passing invalid bytecode?